### DISCUSSION TITLE
Backend Source

### DISCUSSION THREAD

**Discussion by shivamCode0** [cbracketdash/AMCapp#38] [General]
I really love this website and have used it a lot to practice for math competitions. It's really cool that you made this open source.

Is it possible for the Cloudflare worker source to be published? I totally understand if it contains sensitive information but can you please share how you compiled the problems together. Thanks!

@Polydynamical

**Comment by shivamCode0** [comment]
Are there any updates on this?

**Comment by andrewboldi** [comment]
Yeah! Just saw your question. Initially, the website was set up to request problems from AoPS and loaded them through a NO-CORS Proxy Protocol to bypass cross-origin website requests. The Cloudflare worker was set up initially to alter the headers of the XHTML request from the javascript source code to correctly display the problems to the user. However, we have recently devised a method of retrieving problems through a scripting language and storing the problems in a private repository-like database. The worker now makes requests to that database with proper formatting and tags for the javascript to subsequently handle.

Glad you like it!
